homeassistant/components/husqvarna_automower_ble/config_flow.py

```diff
@@ -116,6 +150,63 @@ async def async_step_user(
             data_schema=vol.Schema(
                 {
                     vol.Required(CONF_ADDRESS): str,
+                    vol.Required(CONF_PIN): str,
+                },
+            ),
+        )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauthentication upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm reauthentication dialog."""
+        errors: dict[str, str] = {}
+
+        if user_input:
+            channel_id = random.randint(1, 0xFFFFFFFF)
+            self.address = user_input[CONF_ADDRESS]
+            self.pin = user_input[CONF_PIN]
+
+            pin = int(self.pin) if self.pin else None
```

Comment:
why `if self.pin`?

Reply:
To avoid running `int()` on a `None`

Reply:
But self.pin can't be None right?

Reply:
Ah, true. It can't now thanks to https://github.com/home-assistant/core/pull/135440#discussion_r2011712571